numba_rvsdg/core/datastructures/ast_transforms.py

```diff
@@ -705,16 +706,23 @@ def rlookup(self, region_block: RegionBlock, item: Any) -> Any:
         else:
             raise KeyError(f"Item {item} not found in subregion or parent")
 
-    def codegen(self, block: Any) -> list[ast.AST]:
+    def codegen(self, block: Any) -> MutableSequence[ast.AST]:
```

Comment:
Instead of casting result of `codegen()`, does it always return `list[ast.stmt]`?

Reply:
no, it may also generate a list with `ast.stmt` and `ast.expr` mixed. I.e. a list of `ast.AST` nodes. But `list` isn't covariant and since both `ast.expr` and `ast.stmt` are subclasses of `ast.AST` we need to use `MutableSequence`. NB: I also tried using `Sequence` but that fails to type the `list.append` correctly.